### PR TITLE
Remove unnecessary annotations from the rest-client doc

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: docs/src/main/asciidoc/rest-client-reactive.adoc [hunk 1]
```diff
@@ -121,14 +121,19 @@ import java.util.Set;
 public interface CountriesService {
 
     @GET
-    @Path("/name/{name}")
-    Set<Country> getByName(@PathParam("name") String name);
+    @Path("/name/{nm}")
+    Set<Country> getByName(String nm);
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest-client-reactive.adoc:125]: Can't we use name here? I really don't see the point in shortening things and making them less clear.

Comment (geoand) [reply docs/src/main/asciidoc/rest-client-reactive.adoc:125]: The reason I did it was to try to not confuse the reader with `name` everywhere and to make it clear which part needs to match

Comment (michalszynkiewicz) [reply docs/src/main/asciidoc/rest-client-reactive.adoc:125]: what about `countryName`?

Comment (geoand) [reply docs/src/main/asciidoc/rest-client-reactive.adoc]: Done

**Review by michalszynkiewicz — APPROVED** [review]

**Comment by michalszynkiewicz** [comment]
@geoand do you plan to update the quickstart too?

**Comment by geoand** [comment]
Certainly, I'll do that later on today

**Comment by geoand** [comment]
Here it is: https://github.com/quarkusio/quarkus-quickstarts/pull/845

**Pull request merged**